contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/util/SimpleHttp.java

```diff
@@ -0,0 +1,235 @@
+
+    try {
+      // Execute the request
+      Response response = client
+        .newCall(request)
+        .execute();
+      logger.debug(response.toString());
+
+      // If the request is unsuccessful, throw a UserException
+      if (!response.isSuccessful()) {
+        throw UserException.dataReadError()
+          .message("Error retrieving data: " + response.code() + " " + response.message())
+          .addContext("Response code: ", response.code())
+          .build(logger);
+      }
+      logger.debug("HTTP Request for {} successful.", urlStr);
+      logger.debug("Response Headers: {} ", response.headers().toString());
+
+      // Return the InputStream of the response
+      return response.body().byteStream();
+    } catch (IOException e) {
+      throw UserException.functionError()
+        .message("Error retrieving data")
+        .addContext(e.getMessage())
+        .build(logger);
+    }
+  }
+
+  /**
+   * Function configures the OkHTTP3 server object with configuration info from the user.
+   *
+   * @return OkHttpClient configured server
+   */
+  private OkHttpClient setupServer() {
+    Builder builder = new OkHttpClient.Builder();
+
+    // Set up the HTTP Cache.   Future possibilities include making the cache size and retention configurable but
+    // right now it is on or off.  The writer will write to the Drill temp directory if it is accessible and
+    // output a warning if not.
+    if (config.cacheResults()) {
+      setupCache(builder);
+    }
+
+    // If the API uses basic authentication add the authentication code.
+    if (apiConfig.authType().toLowerCase().equals("basic")) {
+      logger.debug("Adding Interceptor");
+      builder.addInterceptor(new BasicAuthInterceptor(apiConfig.userName(), apiConfig.password()));
+    }
+
+    // Set timeout
+    builder.connectTimeout(config.timeout(), TimeUnit.SECONDS);
+    builder.writeTimeout(config.timeout(), TimeUnit.SECONDS);
+    builder.readTimeout(config.timeout(), TimeUnit.SECONDS);
+
+    return builder.build();
+  }
+
+  /**
+   * This function accepts a Builder object as input and configures response caching. In order for
+   * caching to work, the DRILL_TMP_DIR variable must be set either as a system environment variable or in the
+   * Drill configurations.
+   * <p>
+   * The function will attempt to get the DRILL_TMP_DIR from these places, and if it cannot, it will issue a warning in the logger.
+   *
+   * @param builder Builder the Builder object to which the cacheing is to be configured
+   */
+  private void setupCache(Builder builder) {
+    int cacheSize = 10 * 1024 * 1024;   // TODO Add cache size in MB to config
+    String drillTempDir;
+
+    try {
+      if (context.getOptions().getOption(ExecConstants.DRILL_TMP_DIR) != null) {
+        drillTempDir = context.getOptions().getOption(ExecConstants.DRILL_TMP_DIR).string_val;
+      } else {
+        drillTempDir = System.getenv("DRILL_TMP_DIR");

```

Comment:
This logic really should not be here. Don't we have a utility to do this somewhere? If not, let's make one. I'm sure we have multiple other places that do this same thing.

Reply:
I did a search in the Drill code for the `DRILL_TMP` variables and this was what I found.  Unfortunately, there isn't a utility.  I'll create a new PR for this.

Reply:
Created a JIRA for this: 
https://issues.apache.org/jira/browse/DRILL-7599